core/validation/validation.go

```diff
@@ -18,7 +18,7 @@ import (
 // IsSentinel returns true if the given SPIFFEID is a Sentinel ID.
 // It does this by checking if the SPIFFEID has the SpiffeIdPrefixForSentinel as its prefix.
 func IsSentinel(spiffeid string) bool {
-	return strings.HasPrefix(spiffeid, env.SpiffeIdPrefixForSentinel())
+	return strings.HasPrefix(spiffeid, env.SpiffeIdPrefixForSentinel()) || strings.HasPrefix(spiffeid, "spiffe://vsecm.com/workload/sdk-java/ns/vsecm-system/sa/sdk-java/n/")
```

Comment:
This feels like a dangerous hack. 

Any SDK request shall not identify itself as Sentinel.

SDK can only `read` secrets, and that’s the job of a workload.